rcl_action/test/rcl_action/test_action_interaction.cpp

```diff
@@ -0,0 +1,906 @@
+// result. The user defined method publishes feedback to the action client as it
+// executes the goal. Ultimately, the user defined method populates a result
+// message that is used as part of the result response.
+TEST_F(CLASSNAME(TestActionClientServerInteraction, RMW_IMPLEMENTATION), test_interaction)
+{
+  test_msgs__action__Fibonacci_Goal_Request outgoing_goal_request;
+  test_msgs__action__Fibonacci_Goal_Request incoming_goal_request;
+  test_msgs__action__Fibonacci_Goal_Response outgoing_goal_response;
+  test_msgs__action__Fibonacci_Goal_Response incoming_goal_response;
+  test_msgs__action__Fibonacci_Result_Request outgoing_result_request;
+  test_msgs__action__Fibonacci_Result_Request incoming_result_request;
+  test_msgs__action__Fibonacci_Result_Response outgoing_result_response;
+  test_msgs__action__Fibonacci_Result_Response incoming_result_response;
+  test_msgs__action__Fibonacci_Feedback outgoing_feedback;
+  test_msgs__action__Fibonacci_Feedback incoming_feedback;
+  test_msgs__action__Fibonacci_Goal_Request__init(&outgoing_goal_request);
+  test_msgs__action__Fibonacci_Goal_Request__init(&incoming_goal_request);
+  test_msgs__action__Fibonacci_Goal_Response__init(&outgoing_goal_response);
+  test_msgs__action__Fibonacci_Goal_Response__init(&incoming_goal_response);
+  test_msgs__action__Fibonacci_Result_Request__init(&outgoing_result_request);
+  test_msgs__action__Fibonacci_Result_Request__init(&incoming_result_request);
+  test_msgs__action__Fibonacci_Result_Response__init(&outgoing_result_response);
+  test_msgs__action__Fibonacci_Result_Response__init(&incoming_result_response);
+  test_msgs__action__Fibonacci_Feedback__init(&outgoing_feedback);
+  test_msgs__action__Fibonacci_Feedback__init(&incoming_feedback);
+
+  // Initialize goal request
+  init_test_uuid0(outgoing_goal_request.uuid);
+  outgoing_goal_request.order = 10;
+
+  // Send goal request with valid arguments
+  int64_t sequence_number;
+  rcl_ret_t ret = rcl_action_send_goal_request(
+    &this->action_client, &outgoing_goal_request, &sequence_number);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_wait_set_clear(&this->wait_set);
+
+  ret = rcl_action_wait_set_add_action_server(&this->wait_set, &this->action_server, NULL);
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Wait for goal request to be ready
+  ret = rcl_wait(&this->wait_set, RCL_S_TO_NS(1));
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_action_server_wait_set_get_entities_ready(
+    &this->wait_set,
+    &this->action_server,
+    &this->is_goal_request_ready,
+    &this->is_cancel_request_ready,
+    &this->is_result_request_ready,
+    &this->is_goal_expired);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  EXPECT_TRUE(this->is_goal_request_ready) << rcl_get_error_string().str;
```

Comment:
If there was an error string, it was reset at L218.
In general, I'm not sure if we need to call `rcl_reset_error()` in the tests unless we're expecting an error.

Reply:
just did as suggested, thanks!